tests/library/inspector/console-api.spec.ts

```diff
@@ -121,3 +121,36 @@ it('expected properties on playwright object', async ({ page }) => {
     'getByRole',
   ]);
 });
+
+
+it('should correcly inject itself into popups', async ({ page }) => {
```

Comment:
This test passes for me 100 out of 100 runs without any changes to playwright. Can you add a test that is failing without your change?

Reply:
I added code to the test to inject the script like we do it in the setting where I encountered the bug. Unfortunately, the init script is not running at all like this and I don't understand why.  

Yes, I know this is not a public API, but we at Octomind need to inject this stuff ourselves, so we have a copy of the injected package in our code and use it like the code I added to the test. 

This is how I encountered this bug (I am 100% sure there is actually a bug here, I am just not sure how to reproduce it in a test scenario here. I can record a video of the bug in action if that helps)